iterable-extension/src/main/java/com/mparticle/ext/iterable/IterableExtension.java

```diff
@@ -589,21 +597,33 @@ public void processPushMessageReceiptEvent(PushMessageReceiptEvent event) throws
                 //Android and iOS have differently encoded payload formats. See the tests for examples.
                 if (event.getRequest().getRuntimeEnvironment() instanceof AndroidRuntimeEnvironment) {
                     Map<String, Object> iterableMap = mapper.readValue((String) payload.get("itbl"), Map.class);
-                    request.campaignId = Integer.parseInt(mapper.writeValueAsString(iterableMap.get("campaignId")));
-                    request.templateId = Integer.parseInt(mapper.writeValueAsString(iterableMap.get("templateId")));
+                    request.campaignId = convertItblPayloadFieldToInt(iterableMap.get("campaignId"));
+                    request.templateId = convertItblPayloadFieldToInt(iterableMap.get("templateId"));
                     request.messageId = (String) iterableMap.get("messageId");
                 } else {
-                    request.campaignId = Integer.parseInt(mapper.writeValueAsString(((Map) payload.get("itbl")).get("campaignId")));
-                    request.templateId = Integer.parseInt(mapper.writeValueAsString(((Map) payload.get("itbl")).get("templateId")));
+                    request.campaignId = convertItblPayloadFieldToInt(((Map) payload.get("itbl")).get("campaignId"));
```

Comment:
I'd also extract the itbl part into iterableMap, like above (although parsing is different from Android here).